### PR TITLE
added unit test for Suite

### PR DESCRIPTION
**Pull request by hirata10**
We want to add a few more unit tests here, I'm planning to package some into this PR.

### CODE REVIEW

**Comment by hirata10** [comment]
I suppose that's true ...
```
tests/pyimcom/test_utils.py:184:10: B017 `pytest.raises(Exception)` should be considered evil
```
I can make this one more specific :)

**Comment by hirata10** [comment]
I want to add more tests, but some of them will require more work and I think it's best to get these in here now.

File: src/pyimcom/wcsutil.py
Comment (nihardalal) [inline src/pyimcom/wcsutil.py:1]: Clean up for these (and above) files looks good to me

File: tests/pyimcom/test_cfg.py
Comment (nihardalal) [inline tests/pyimcom/test_cfg.py:1]: Looks good!

File: tests/pyimcom/test_cprs.py [hunk 1]
```diff
@@ -122,3 +144,36 @@ def test_allcprs(tmp_path):
...
+    # compression dictionary when there's nothing
+    hdulist = fits.HDUList([fits.PrimaryHDU(x), fits.ImageHDU(x[:2, :], name="X")])
+    d = CompressedOutput.get_compression_dict(hdulist, 1)
+    assert len(d) == 0
+
+    # I24Cube errors
+    p = {"VMIN": -1.0, "VMAX": 1.0}
+    with pytest.raises(ValueError):
+        I24Cube(x, {"BITKEEP": 33} | p)
+    with pytest.raises(ValueError):
```
Comment (nihardalal) [inline tests/pyimcom/test_cprs.py:171]: Why should this call raise a ValueError? Not clear to me on an immediate pass of the code.

File: tests/pyimcom/test_pyimcom.py [hunk 2]
```diff
@@ -842,8 +870,8 @@ def test_PyIMCOM_run1(tmp_path, setup):
         # values from noise layers
         test5 = np.array([[0.7601451, 0.9042513], [0.64049757, 0.70962816]])
         test6 = np.array([[0.24921854, -0.23588116], [-0.39272013, -0.6111549]])
-        assert np.amax(np.abs(fblock[0].data[0, 5, :2, :2] - test5)) < 1e-3
-        assert np.amax(np.abs(fblock[0].data[0, 6, :2, :2] - test6)) < 1e-3
+        assert np.amax(np.abs(fblock[0].data[0, 5, :2, :2] - test5)) < 1.5e-3
```
Comment (nihardalal) [inline tests/pyimcom/test_pyimcom.py:873]: Why did these get relaxed?

File: tests/pyimcom/test_updatecube.py
Comment (nihardalal) [inline tests/pyimcom/test_updatecube.py:1]: I think this looks okay to me - I admittedly only gave it a cursory look because the file i/o is hard to parse.

File: tests/pyimcom/test_utils.py [hunk 3]
```diff
@@ -159,3 +169,27 @@ def test_percentiles_and_delete(tmp_path):
...
+def test_resolve_bounds():
+    """Edge cases in resolve_bounds."""
+
+    a = resolve_bounds(None, 36)
+    assert a == (0, 36, 0, 36)
+
+    a = resolve_bounds("this_isn't_a_valid_anything", 30)
+    assert a == (0, 30, 0, 30)
+
+    # these should create errores
```
Comment (nihardalal) [inline tests/pyimcom/test_utils.py]:
```suggestion
    # these should create errors
```

File: tests/pyimcom/test_wcs.py [hunk 4]
```diff
@@ -69,4 +75,44 @@ def test_fits(tmp_path):
...
+    # pipeline version of the WCS
+    distortion = models.AffineTransformation2D(cd, translation=[0, 0])
+    distortion.inverse = models.AffineTransformation2D(np.linalg.inv(cd), translation=[0, 0])
+    celestial_rotation = models.RotateNative2Celestial(9.55, -41.0, 180.0)
+    shift = models.Shift(-(crpix[0] - 1)) & models.Shift(-(crpix[1] - 1))
+    det2sky = shift | distortion | models.Pix2Sky_ARC() | celestial_rotation
+    det2sky.name = "TestMapping1"
+    detector_frame = cf.Frame2D(name="detector", axes_names=("x", "y"), unit=(u.pix, u.pix))
+    sky_frame = cf.CelestialFrame(reference_frame=coord.ICRS(), name="icrs", unit=(u.deg, u.deg))
+    sca_gwcs = gwcs.WCS([(detector_frame, det2sky), (sky_frame, None)])
```
Comment (nihardalal) [inline tests/pyimcom/test_wcs.py:99]: Looks fine to me, although I didn't know that we had to take this many steps to do a coordinate transformation.

**Review by nihardalal — APPROVED** [review]
A few minor comments here and there, but all the tests look okay to me, so feel free to merge.

File: tests/pyimcom/test_cprs.py (hunk 1 above)
Comment (hirata10) [reply tests/pyimcom/test_cprs.py:171]: This compression algorithm only supports 24 bits, so if you request more you should get an error.

File: tests/pyimcom/test_wcs.py (hunk 4 above)
Comment (hirata10) [reply tests/pyimcom/test_wcs.py:99]: It's just to put it in the gwcs representation (as opposed to FITS headers, which have a similar amount of data, but a lot of this piping is hidden in astropy).

File: tests/pyimcom/test_pyimcom.py (hunk 2 above)
Comment (hirata10) [reply tests/pyimcom/test_pyimcom.py:873]: I'll check specifically which import update caused this to fail on my machine.

File: tests/pyimcom/test_cprs.py (hunk 1 above)
Comment (nihardalal) [reply tests/pyimcom/test_cprs.py:171]: Sorry, I meant the ravel one below, but no big deal
